lib/dataModel.py

```diff
@@ -0,0 +1,198 @@
+        self.drm = None
+        self.measurements = []
+        self.rightsURI = None
+        self.link = None
+
+        if (isinstance(link, Link)):
+            self.link = link
+        else:
+            self.setLink(url=link)
+
+    def setLink(self, **linkFields):
+        newLink = Link.createFromDict(**linkFields)
+        self.link = newLink
+
+
+class Agent(DataObject):
+    def __init__(self, name=None, role=None, aliases=None, birth=None, death=None, link=None):
+        super()
+        self.name = name
+        self.sortName = None
+        self.lcnaf = None
+        self.viaf = None
+        self.biography = None
+        self.aliases = aliases
+        self.birthDate = birth
+        self.deathDate = death
+        self.link = link
+
+        if isinstance(role, (str, int)):
+            self.roles = [role]
+        else:
+            self.roles = role
+
+    # TODO This method is pretty ugly and there must be a better way to merge
+    # agent records. However, it does work
+    @staticmethod
+    def checkForMatches(newAgents, agents):
+        merged = defaultdict(dict)
+        for agent in agents:
+            merger = list(filter(lambda x: jaro_winkler(x['name'].lower(), agent['name'].lower()) > 0.8, newAgents))
+            if(len(merger) > 0):
+                mergedAgent = merger[0]
+                merged[mergedAgent['name']] = Agent.mergeFromDict(agent, mergedAgent)
+            else:
+                merged[agent.name] = agent
+
+        for newAgent in newAgents:
+            if newAgent.name not in merged:
+                merged[newAgent.name] = newAgent
+
+        return merged.values()
+
+    @staticmethod
+    def mergeFromDict(otherAgent, agent):
+        if isinstance(otherAgent, Agent):
+            otherAgent = otherAgent.getDictValue()
+        for key, value in otherAgent.items():
+            if key == 'aliases' and agent.aliases is not None :
+                agent['aliases'].extend(value)
+                continue
+            if key == 'roles':
+                if isinstance(value, (str, int)):
+                    value = [value]
+                agent['roles'].extend(value)
+            if agent[key] is None:
+                agent[key] = value
+        return agent
+
+
+class Identifier(DataObject):
+    def __init__(self, type=None, identifier=None, weight=None):
+        super()
+        self.type = type
+        self.identifier = identifier
+        self.weight = weight
+
+
+class Link(DataObject):
+    def __init__(self, url=None, mediaType=None, relType=None):
+        super()
+        self.url = url
+        self.mediaType = mediaType
+        self.content = None
+        self.relType = None
+        self.thumbnail = None
+
+
+class Subject(DataObject):
+    def __init__(self, subjectType=None, value=None, weight=None):
+        super()
+        self.type = subjectType
+        self.identifier = None
+        self.value = value
+        self.weight = weight
+        self.measurements = []
```

Comment:
I'm not sure what the measurements would be here since you already have `weight`.

Reply:
I'm not sure either. I know I had an idea of some values that we might want to store there, but I can't recall what they were, so that should probably come out.